ReplayKitExample/BroadcastExtension/SampleHandler.swift

```diff
@@ -0,0 +1,229 @@
+
+        TwilioVideo.audioDevice = ExampleCoreAudioDevice(audioCapturer: self)
+
+        // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+        let accessToken = "TWILIO-ACCESS-TOKEN";
+
+        let localScreenTrack = TVILocalVideoTrack(capturer: self)
+        let h264VideoCodec = TVIH264Codec()
+        let localAudioTrack = TVILocalAudioTrack()
+        let connectOptions = TVIConnectOptions.init(token: accessToken) { (builder) in
+
+            // Use the local media that we prepared earlier.
+            builder.audioTracks = [localAudioTrack!]
+            builder.videoTracks = [localScreenTrack!]
+
+            // Use the preferred video codec
+            builder.preferredVideoCodecs = [h264VideoCodec] as! [TVIVideoCodec]
+
+            // The name of the Room where the Client will attempt to connect to. Please note that if you pass an empty
+            // Room `name`, the Client will create one for you. You can get the name or sid from any connected Room.
+            if #available(iOS 12.0, *) {
+                builder.roomName = "test"
+            } else {
+                builder.roomName = setupInfo?["RoomName"] as? String
+            }
+        }
+
+        // Connect to the Room using the options we provided.
+        room = TwilioVideo.connect(with: connectOptions, delegate: self)
+
+        // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+        print("broadcastStarted")
+    }
+
+    override func broadcastPaused() {
+        // User has requested to pause the broadcast. Samples will stop being delivered.
+    }
+
+    override func broadcastResumed() {
+        // User has requested to resume the broadcast. Samples delivery will resume.
+    }
+
+    override func broadcastFinished() {
+        // User has requested to finish the broadcast.
+    }
+
+    override func processSampleBuffer(_ sampleBuffer: CMSampleBuffer, with sampleBufferType: RPSampleBufferType) {
+        RPScreenRecorder.shared().isMicrophoneEnabled = true
+        switch sampleBufferType {
+        case RPSampleBufferType.video:
+            if (captureConsumer != nil) {
+                processVideoSampleBuffer(sampleBuffer)
+            }
+            break
+
+        case RPSampleBufferType.audioApp:
+            // TODO: Mix app's audio and microphone audio at audioDevice's capturing sample rate and supply it.
+
+            /*
+             * TODO: For now, since we are providing app's audio to the audio device here, add an assertion if the
+             * sample rate does not match with audioDevice's audio captureing sample rate.
+             */
+            ExampleCoreAudioDeviceRecordCallback(sampleBuffer)
+            break
+
+        case RPSampleBufferType.audioMic:
+            ExampleCoreAudioDeviceRecordCallback(sampleBuffer)
+            break
```

Comment:
For the first version of the PR, lets focus on one audio source (the mic) at a time. Future iterations can deal with mixing concerns.